### DISCUSSION TITLE
Assets Link Setting for Frontmatter Config.

### DISCUSSION THREAD

**Discussion by louiss0** [estruyf/vscode-front-matter#692] [Ideas]
As you know we have the `"frontMatter.content.publicFolder": "[[workspace]]/apps/blog/src",` setting. 

But when it's used the link that is produced is the link inside of the folder the folder is not included in the path.
That setting is good for unbundled assets but it's not good for assets that are bundled or ones that people use.

## Context 

When using the public folder I like to be specific about what assets belong where so I use a`assets/images/` to put my images.
The `assets/` folder is used to hold all of my bundled assets. The images folder is used for images. When I link to the file in the assets folder `/images/image.jpg` is shown not `/assets/images/image.jpg`.

## Solution 

**I think there should be a  `"frontMatter.content.assetsFolder": "",` setting.**

When it comes to the public folder its name is not supposed to be referred to at all. But bundled files must be accessed. using the folder's name.

**Comment by estruyf** [comment]
Thanks, @louiss0, for starting this discussion. In the latest version, there is a new option for this setting. You can now specify if you want to add relative paths: https://frontmatter.codes/docs/settings/overview#frontmatter.content.publicfolder - we've added this to support the new `assets` folder location of Astro, but it can be used for other projects as well.

In your case, it would be:

```json
{
  "frontMatter.content.publicFolder": {
    "path": "[[workspace]]/apps/blog/src",
    "relative": true
  }
}
```

Would that solve your issue?

**Reply by louiss0** [reply]
I think a better option would be to append the name. 

```json
{
  "frontMatter.content.publicFolder": {
    "path": "[[workspace]]/apps/blog/src",
    "appendFolderName": true
  }
}
```

**Comment by louiss0** [comment]
Good job. I think the word relative would make people think a relative path is being created `./` but that's not the case.

**Reply by estruyf** [reply]
It creates a relative path from where the content lives, but if the assets are outside the content folder, the full path will be added. That is why I thought it could be a solution.

**Comment by louiss0** [comment]
Thanks. I have now I have another problem its regarding to Categories. I can only use this field only if it's a part of the categories field.
I want to be able to set the categories field the same way I can the `isPreviewImage` property.

**Reply by estruyf** [reply]
Answered on the other discussion you've opened.
